helix-term/src/ui/picker.rs

```diff
@@ -344,11 +347,17 @@ impl<T: Item + 'static> Component for FilePicker<T> {
 
 #[derive(PartialEq, Eq, Debug)]
 struct PickerMatch {
-    index: usize,
     score: i64,
+    index: usize,
     len: usize,
 }
 
+impl PickerMatch {
+    fn key(&self) -> impl Ord {
+        (cmp::Reverse(self.score), self.index, self.len)
```

Comment:
Interestingly, this also fixes the `Eq` impl -- previously the derived impl would disagree with the `Ord`

Reply:
I suppose that was a small oversight in the previous implementation. We probably should have returned `None` in case of equality to be consistent with `PartialEq` so this is definitly more consistent :+1: